### PR TITLE
Move some changes related to RC

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Move some changes from feature/4.4.0_rc_functionality related to the next components:

- Utils
- Telemetry monitor
- SmartObjects
- Secutiry Manager
- Media Manager
- Config profile
- Formatters
- Connection handler

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@AKalinich-Luxoft @AByzhynar @dev-gh please review

File: src/components/include/utils/timer_thread.h [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+  if (wait_seconds_left <
+      std::numeric_limits<int32_t>::max() / milliseconds_in_second) {
+    wait_milliseconds_left = milliseconds_in_second * wait_seconds_left;
+  }
+  return wait_milliseconds_left;
+}
+
+}  // namespace timer
+
+#endif  // SRC_COMPONENTS_INCLUDE_UTILS_TIMER_THREAD_H_
```
Comment (LuxoftAKutsan) [inline src/components/include/utils/timer_thread.h]: @LitvinenkoIra please remove this file and try to check build without it. It looks like redundant

File: src/components/telemetry_monitor/CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +71,14 @@ add_library("TelemetryMonitor" ${SOURCES})
 target_link_libraries("TelemetryMonitor" ${LIBRARIES})
 add_dependencies("TelemetryMonitor" HMI_API MOBILE_API)
 
+set (LIBRARIES
+  HMI_API
+)
+
+add_library("TimeTester" ${SOURCES})
```
Comment (LuxoftAKutsan) [inline src/components/telemetry_monitor/CMakeLists.txt]: @LitvinenkoIra please rename library to TelemetryMonitor

File: src/components/include/security_manager/crypto_manager.h [hunk 3]
```diff
@@ -72,6 +72,9 @@ class CryptoManager : public policy::PolicyHandlerObserver {
   * \return pointer to crypto manager settings class
   */
   virtual const CryptoManagerSettings& get_settings() const = 0;
+#ifdef BUILD_TESTS
+  virtual void SetCertFutureExpTime(const std::string& time) = 0;
```
Comment (LuxoftAKutsan) [inline src/components/include/security_manager/crypto_manager.h]: @LitvinenkoIra remoe this method. It is not used anywhere except tests

File: src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc [hunk 4]
```diff
@@ -43,7 +43,7 @@ namespace test {
 namespace components {
 namespace hmi_message_handler_test {
 
-using ::testing::ReturnRef;
+using ::testing::Return;
```
Comment (LuxoftAKutsan) [inline src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc:46]: @LitvinenkoIra this commit is related to what commit? squash it please.

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

**Review by dev-gh — APPROVED** [review]

**Review by AKalinich-Luxoft — APPROVED** [review]

File: src/components/include/utils/timer_thread.h (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/include/utils/timer_thread.h]: @LuxoftAKutsan Done

File: src/components/telemetry_monitor/CMakeLists.txt (hunk 2 above)
Comment (LitvinenkoIra) [reply src/components/telemetry_monitor/CMakeLists.txt]: @LuxoftAKutsan Done
b08e641ae22acb3f88b40938af9499987261f24a

File: src/components/include/security_manager/crypto_manager.h (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/include/security_manager/crypto_manager.h]: @LuxoftAKutsan Done

File: src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc (hunk 4 above)
Comment (LitvinenkoIra) [reply src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc:46]: @LuxoftAKutsan Done
598c2710ad90accacdc5fbabd094af12f6693a44

**Review by AByzhynar — APPROVED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**